### PR TITLE
fix(deps): Update dependencies (Kotlin 1.6)

### PR DESCRIPTION
**Pull request by Filmaluco**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -113,22 +112,22 @@ dependencies {
...
     // Dependency Injection
-    api 'com.google.dagger:dagger:2.37'
-    kapt 'com.google.dagger:dagger-compiler:2.28.1'
+    api 'com.google.dagger:dagger:2.42'
+    kapt 'com.google.dagger:dagger-compiler:2.42'
 
     // Awala
-    implementation 'tech.relaycorp:awala:1.48.5'
+    implementation 'tech.relaycorp:awala:1.65.0'
     implementation 'tech.relaycorp:cogrpc:1.1.11'
```
Comment (Filmaluco) [inline app/build.gradle:128]: needs a higher version once kotlin is also updated

File: app/build.gradle [hunk 2]
```diff
@@ -137,7 +136,7 @@ dependencies {
 
     // Preferences
     implementation 'androidx.preference:preference:1.1.1'
```
Comment (Filmaluco) [inline app/build.gradle:138]: needs a higher version once kotlin is also updated

File: build.gradle [hunk 3]
```diff
@@ -2,32 +2,34 @@
...
+        kotlinCoroutinesVersion = '1.4.2'
+        grpcVersion = '1.46.0'
         protobufVersion = '3.12.2'
         protobufGradleVersion = '0.8.12'
-        nettyVersion = '4.1.59.Final'
-        aboutLibsVersion = '8.3.1'
-        junitVersion = '5.7.2'
-        roomVersion = '2.3.0'
+        nettyVersion = '4.1.77.Final'
+        aboutLibsVersion = '8.6.3'
```
Comment (Filmaluco) [inline build.gradle:11]: needs a higher version once kotlin is also updated

File: build.gradle [hunk 4]
```diff
@@ -2,32 +2,34 @@
...
+        junitVersion = '5.8.2'
+        roomVersion = '2.4.0-beta02'
```
Comment (Filmaluco) [inline build.gradle:13]: needs a higher version once kotlin is also updated

**Comment by gnarea** [comment]
BTW @Filmaluco, have you tested this branch along with the other PRs for the gateway and ping?

**Comment by Filmaluco** [comment]
> BTW @Filmaluco, have you tested this branch along with the other PRs for the gateway and ping?

No. Courier is the one I have not tested. I plan on doing that today

**Review by gnarea — DISMISSED** [review]
Obrigado! We now just have to wait for GH to fix the issue with CI.

**Review by gnarea — APPROVED** [review]

**Review by gnarea — CHANGES_REQUESTED** [review]
@Filmaluco, now that CI is working again it's reporting an issue with `AppTest`.

**Comment by gnarea** [comment]
@Filmaluco, I just noticed that the instrumentation test is failing on Android 6, not 8 (I thought API 23 = Android 8 🤦🏾‍♂️). I don't have a physical Android 6 device (I only have 5, 8, 11 and 12).

I just pushed a change to use a physical Nexus 5 (Android 6) instead of the virtual one we've been using. If that still fails for the same reason, please go ahead and disable the test on Android 6 as discussed on Slack.

Also, can you please undo the changes you made to try to make the test work on Firebase Test Lab (but didn't work)?

On that note, I noticed you introduced some changes to the Gradle config related to Orchestrator but you didn't enable the Orchestrator in `firebase-test-lab.yml` so those settings have no effect. We should probably enable the Orchestrator anyway in a separate PR though.

**Comment by gnarea** [comment]
The physical Nexus 5 didn't fix the issue :( I'm going to make one more attempt with a non-Nexus device on API 23 and if that doesn't fix it, then let's definitely disable the test on API 23.

On the other hand, more bad news: I discovered that Firebase Test Lab wasn't actually running the tests on Android 5, so I fixed it in master (#532) but this branch is causing the app to crash before the tests run. Can you please look into it? Don't spend too much time on this because I can actually run the same APK on a physical Nexus 4 (Android 5) with no issues whatsoever... So if we can't get to the bottom of this issue within an hour or so, we should just disable Android 5 on Firebase Test Lab.

**Comment by sdsantos** [comment]
@gnarea talked with @Filmaluco and dropped all changes after bumping cogrpc

**Comment by sdsantos** [comment]
> On the other hand, more bad news: I discovered that Firebase Test Lab wasn't actually running the tests on Android 5, so I fixed it in master (#532) but this branch is causing the app to crash before the tests run. Can you please look into it? Don't spend too much time on this because I can actually run the same APK on a physical Nexus 4 (Android 5) with no issues whatsoever... So if we can't get to the bottom of this issue within an hour or so, we should just disable Android 5 on Firebase Test Lab.

Investigated for a bit. The issue is reproducible locally, but there aren't many clues to go on. We only get an easy-to-miss log message: `libc: Fatal signal 11 (SIGSEGV), code 1, fault addr 0x28 in tid 8135 (roidJUnitRunner)`. So a native crash inside the AndroidJUnitRunner. Next steps would be changing/disabling things until we get a different behaviour happening, or try to get a assemble stack trace and go from there. All more than 1 hour tasks 😞

**Review by gnarea — APPROVED** [review]
Thanks @sdsantos! Sorry, I missed your reply 😬 

Sounds good. So now I'm going to:

1. Remove Android 5 and 6 from the test matrix on Firebase Test Lab.
2. Create an issue to try to make it work again in the future.
3. Fix the GitHub checks in CI because there's a required check that's blocking this PR, but it shouldn't be.

Then I'll merge this PR.

**Review by gnarea — APPROVED** [review]

**Pull request merged**